app/routes.py

```diff
@@ -1,2 +1,446 @@
-from flask import Blueprint
+from flask import Blueprint, make_response, jsonify, request
+from app import db
+from app.models.video import Video
+from app.models.customer import Customer
+from app.models.rental import Rental
+from datetime import datetime
+import re
 
+
+customers_bp = Blueprint("customers", __name__, url_prefix="/customers")
+videos_bp = Blueprint("videos", __name__, url_prefix="/videos")
+rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+# ----------------- helper functions ------------------------
+
+
+def is_number(input):
+    condition = isinstance(input, int) or isinstance(input, float)
+    return True if condition else False
+
+
+def is_valid_phone(input):
+    """Validates the phone number pattern (908) 949-6758"""
+    pattern = re.compile(r'\(\d{3}\) \d{3}-\d{4}')
+    return re.match(pattern, input) is not None
+
+
+# -------------- CRUD for /customers ------------------------
+
+
+@customers_bp.route("", methods=["GET"])
+def get_all_customers():
+    """Lists all existing customers and details about each customer."""
+    customers = Customer.query.all()
+    response = [customer.to_json() for customer in customers]
+    return jsonify(response), 200
+
+
+@customers_bp.route("/<customer_id>", methods=["GET"])
+def get_customer_by_id(customer_id):
+    """Gives back details about specific customer."""
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return make_response({
+            "errors": [
+                "Not Found",
+                "Customer does not exist"
+            ]
+        }, 404)
```

Comment:
This works great! We could combine these lines into one using `get_or_404`. This can be used in place of `get` for any query made.

```suggestion
    customer = Customer.query.get_or_404(customer_id)
```

Reply:
How do I leave an error message in that case?